uvicorn/protocols/http/h11_impl.py

```diff
@@ -188,6 +188,7 @@ def handle_events(self):
                 raw_path, _, query_string = event.target.partition(b"?")
                 self.scope = {
                     "type": "http",
+                    "asgi": {"version": "2.1", "spec_version": "2.1"},
```

Comment:
It seems that `scope["asgi"]["version"]` actually refers to either ASGI 2 or ASGI 3, i.e. the [ASGI spec](https://github.com/django/asgiref/blob/master/specs/asgi.rst):

> The key `scope["asgi"]` will also be present as a dictionary containing a `scope["asgi"]["version"]` key that corresponds to the ASGI version the server implements. If missing, the version should default to "2.0".

(OTOH `spec_version` refers to the ASGI [HTTP/WebSocket spec](https://github.com/django/asgiref/blob/master/specs/www.rst).)

We support both versions, and which one is used depends on the app callable interface - that information is available as `config.interface`.

So adding this to `Config`…

```python
class Config:
    @property
    def asgi_version(self) -> str:
        return {"asgi2": "2.0", "asgi3": "3.0"}[self.interface]
```

We should be able to use:

```suggestion
                    "asgi": {"version": config.asgi_version, "spec_version": "2.1"},
```

Reply:
done